public/docs/_examples/dependency-injection/dart/lib/heroes/hero_service_provider.dart

```diff
@@ -0,0 +1,18 @@
+// #docregion
+
+import "package:angular2/core.dart";
+import "hero_service.dart";
+import "../logger_service.dart";
+import "../user_service.dart";
+
+// #docregion factory
+heroServiceFactory(Logger logger, UserService userService) =>
```

Comment:
I guess not. This is something done outside the class (3rd-party) otherwise the DI factory wouldn't   be necessary in the first place.